chia/wallet/wallet_node_api.py

```diff
@@ -97,6 +97,12 @@ async def transaction_ack(self, ack: wallet_protocol.TransactionAck, peer: WSChi
         async with self.wallet_node.wallet_state_manager.lock:
             assert peer.peer_node_id is not None
             name = peer.peer_node_id.hex()
+            if peer.peer_node_id in self.wallet_node._tx_messages_in_progress:
+                self.wallet_node._tx_messages_in_progress[peer.peer_node_id] = [
+                    txid for txid in self.wallet_node._tx_messages_in_progress[peer.peer_node_id] if txid != ack.txid
+                ]
+                if self.wallet_node._tx_messages_in_progress[peer.peer_node_id] == []:
+                    del self.wallet_node._tx_messages_in_progress[peer.peer_node_id]
```

Comment:
the responsibility to add items to this list is `wallet_node.py` and the responsibility to remove them is in `wallet_node_api.py`. It might be a bit risky. e.g. there might be an edge case where items to be left in the dict indefnitely

Reply:
I think I would view the wallet_node_api as an extension of the wallet node.  The risk comes from the asynchronicity and I'm not sure there's necessarily a way around that except maybe to clear it periodically.